modules/core/arrow-core/src/test/kotlin/arrow/core/extensions/ListKExtensionTest.kt

```diff
@@ -0,0 +1,33 @@
+package arrow.core.extensions
+
+import arrow.Kind
+import arrow.core.ForListK
+import arrow.core.Ior
+import arrow.core.ListK
+import arrow.core.extensions.listk.eq.eq
+import arrow.core.extensions.listk.foldable.foldable
+import arrow.core.extensions.listk.semialign.semialign
+import arrow.core.k
+import arrow.test.UnitSpec
+import arrow.test.generators.listK
+import arrow.test.laws.SemialignLaws
+import arrow.typeclasses.Eq
+import io.kotlintest.properties.Gen
+import io.kotlintest.shouldBe
+
+class ListKExtensionTest : UnitSpec() {
+  init {
+    testLaws(SemialignLaws.laws(ListK.semialign(),
+      Gen.listK(Gen.int()) as Gen<Kind<ForListK, Int>>,
+      { ListK.eq(it) as Eq<Kind<ForListK, *>> },
+      ListK.foldable()
+    ))
+
+    "lists with different lengths are padded" {
+      ListK.semialign().align(
```

Comment:
Please use property based testing instead:
```
// pseudo code like stuff
forAll (genList, genList) { l1, l2 ->
 align(l1, l2).size == max(l1.size, l2.size)
}
```
And similar `align(l1, l2).take(max(l1.size, l2.size)).all { isBoth }` and similarly check the rest for left or right

Reply:
i changed the custom tests to use property based testing

Reply:
You can also test that everything from `min(s1, s2)` is either left or right in a similar manner. Maps and sequences should also have tests like that. Once that is done the tests are good to go i think :+1:

Reply:
i have added more tests for ListK, MapK and SequenceK that should cover these cases now